analyzers/tests/SonarAnalyzer.UnitTest/TestCases/CheckArgumentException.CSharp9.cs

```diff
@@ -71,3 +71,18 @@ public string Request
         }
     }
 }
+
+// https://github.com/SonarSource/sonar-dotnet/issues/4423
+public class Repro_4423
+{
+    public void InsideLocalFunction_Static(string methodArg)
+    {
+        Something(null);
+
+        static void Something(string localArg)
+        {
+            throw new ArgumentNullException(nameof(localArg));   // Compliant
```

Comment:
In C# 9 we can have also top level functions. I think it would be great to have a test case to cover that.